In [1]:
import pandas as pd
from neuronunit.optimization.optimization_management import public_opt
from sciunit.scores import RelativeDifferenceScore, ZScore
from sciunit.utils import config_set, config_get
config_set('PREVALIDATE', False)
from neuronunit.allenapi.allen_data_driven import make_allen_hard_coded_limited, make_allen_hard_coded_complete
import warnings
from neuronunit.optimization.model_parameters import BPO_PARAMS
warnings.filterwarnings("ignore")
_,_,_,a_cells = make_allen_hard_coded_limited()


[
    allen_suite_623960880,
    allen_suite_623893177,
    allen_suite471819401,
    allen_suite482493761,
    a_cells2,
    df
] = make_allen_hard_coded_complete()



/home/user/anaconda3/lib/python3.7/site-packages/cerberus/validator.py:1609: UserWarning: No validation schema is defined for the arguments of rule 'not_zero_obs_zscore'
  "'%s'" % method_name.split('_', 2)[-1]
test


# In this notebook
we fit Izhikevich and Adaptive Exponential Integrate and fire models to electrophysiology observations 
from the Allen Brain, using BluePyOpt and NeuronUnit.

# Immediately Below 
* is a list of some of the observations we will use:

In [2]:
df

TimeConstantTest       name RestingPotentialTest  \
623960880          23.8 ms  623960880             -65.1 mV   
623893177          27.8 ms  623893177             -77.0 mV   
471819401          13.8 ms  471819401             -77.5 mV   
482493761          24.4 ms  482493761             -71.6 mV   

          InputResistanceTest RheobaseTest  
623960880       241.0 megaohm      70.0 pA  
623893177       136.0 megaohm     190.0 pA  
471819401       132.0 megaohm     190.0 pA  
482493761       132.0 megaohm      70.0 pA

Allen Specimen ID: 471819401
Four data fitting constraints:
* RheobaseTest	
* TimeConstantTest	
* RestingPotentialTest	
* InputResistanceTest	
# Model AdExp

In [ ]:
opt,obs_preds,df = public_opt(
    a_cells['471819401'],
    BPO_PARAMS,
    '471819401',
    "ADEXP",
    100,
    200,
    "IBEA",
    score_type=RelativeDifferenceScore
    )


gen	nevals	avg    	std    	min    	max 
1  	100   	1538.18	1514.29	12.4581	4000


2  	45    	1495.88	1478.09	12.4581	4000
3  	8     	521.317	693.355	12.4581	4000
4  	9     	169.522	361.724	11.7822	1010.05
5  	7     	345.401	468.199	11.7822	1009.93
6  	6     	558.644	494.769	11.7822	1007.78
7  	9     	597.779	489.112	9.83394	1007.67
8  	7     	421.771	488.943	9.83394	1006.98
9  	11    	343.136	469.018	9.74488	1008.07
10 	14    	421.319	488.847	9.74488	1007.56
11 	11    	440.64 	491.915	8.8705 	1008.69
12 	5     	264.488	432.584	8.8705 	1006.71
13 	10    	536.616	495.672	5.4392 	1007.94
14 	8     	438.925	492.656	5.4392 	1007.94
15 	6     	185.129	379.489	5.4392 	1006.17
16 	10    	244.933	421.679	5.4392 	1007.21
17 	7     	418.84 	490.367	5.4392 	1005.93
18 	12    	419.517	490.006	5.4392 	1007.43
19 	12    	536.703	497.164	5.4392 	1008.4 
20 	12    	341.273	469.346	5.4392 	1007.4 
21 	6     	183.519	380.349	5.4392 	1006.2 
22 	12    	380.979	481.393	5.4392 	1008.41
23 	15    	360.138	476.521	5.4392 	1007.95
24 	12    	223.606	409.826	5.4392 	1006.6 
25 	17    	457.52

In [ ]:
df

# Solution Model Parameters:


In [ ]:
opt.attrs

Allen Specimen ID: 471819401
Four data fitting constraints:
* RheobaseTest	
* TimeConstantTest	
* RestingPotentialTest	
* InputResistanceTest	
# Model Izhikevich

In [ ]:
opt,obs_preds,df = public_opt(
    a_cells['471819401'],
    BPO_PARAMS,
    'fi_curve',
    "IZHI",
    100,
    360,
    "IBEA",
    )

In [ ]:
df

# Solution Parameters:

In [ ]:
opt.attrs

# Just FI curve on AdExp model
Allen Specimin Id: 471819401

In [ ]:
opt,obs_preds,df = public_opt(
    a_cells['fi_curve'],
    BPO_PARAMS,
    'fi_curve',
    "ADEXP",
    100,
    100,
    "IBEA"
    )

In [ ]:
df


In [ ]:
opt.attrs

Allen Specimen ID: 623960880
Four data fitting constraints:
* RheobaseTest	
* TimeConstantTest	
* RestingPotentialTest	
* InputResistanceTest	
# Model Izhikevich

In [ ]:

opt,obs_preds,df = public_opt(
    allen_suite_623960880.tests,
    BPO_PARAMS,
    allen_suite_623960880.name,
    "IZHI",
    100,
    100,
    "IBEA"
    )

In [ ]:
df

In [ ]:
opt.attrs